### ISSUE TITLE
It's always timeout when downloading  a larget attachment more than 25M even in a local network.

### ISSUE THREAD

**Issue by znsoftm** [mickstar/Zoo-For-Zotero#122]
I setup a webdav server in my local network, when I download an attachment more than 25M, it always complains time is out for downloading.

Can you add a setup item to change the value of timeout ?

**Comment by smatthie** [comment]
I have the same problem. It looks like the program always raises a timeout errors afte 30 seconds even when it is still downloading.
It should only timeout when it hasn't received any data for a period, not after a fixed time from the start.

**Comment by pstaerk** [comment]
I also have experienced the same problem.

This seems to be the line that needs changing:
https://github.com/mickstar/Zoo-For-Zotero/blob/1e9f59c5098ce649fd9119ec8bb4828cc68055d6/app/src/main/java/com/mickstarify/zooforzotero/ZoteroAPI/Webdav.kt#L204

**Comment by Madcliff** [comment]
Yeah I am having this issue as well. Most of my pdfs cannot be downloaded because of this...

**Comment by robcnewman** [comment]
Also experiencing this issue, has anybody got a solution?

**Comment by qldaisd** [comment]
Same question. Is there any solution?

**Comment by hughplay** [comment]
I found the author has already implemented the feature for customizing the timeout. I tested the new branch [feature/upgrade-webdav](https://github.com/mickstar/Zoo-For-Zotero/tree/feature/upgrade-webdav) by building the project with AndroidStudio.

The only extra thing that is not included in the source code is a `google-services.json` file, which can be obtained by [following this page](https://firebase.google.com/docs/android/setup).

It worked fine on my phone with extra custom configurations:

<image src="https://user-images.githubusercontent.com/30553791/225528717-5b0a9513-94a1-4683-8b9b-738d27740cd9.png" width=240>

Many thanks to the author for his great efforts.

**Comment by rafadscruz** [comment]
Any chance of having this update soon? I tried to build it in Android Studio, but could not connect during the login.
